### PR TITLE
Add deadline to token claim

### PR DESCRIPTION
**Pull request by herbig**
## Description

Adds an optional deadline to the token claim contract, with a value of 0 being indefinite

## Issue / Notion doc (if applicable)
closes https://github.com/decent-dao/fractal-contracts/issues/29

### CODE REVIEW

File: contracts/TokenClaim.sol [hunk 1]
```diff
@@ -62,4 +67,13 @@ contract TokenClaim is FactoryFriendly, ITokenClaim {
             (VotesToken(parentToken).balanceOfAt(claimer, snapShotId) * parentAllocation) /
             VotesToken(parentToken).totalSupplyAt(snapShotId);
     }
+
+    /// @notice Returns unclaimed tokens after the deadline to the funder.
+    function reclaim() external {
+        require(msg.sender == funder, "caller is not the funder");
+        require(deadline != 0, "no deadline set");
+        require(block.number >= deadline, "deadline has not elapsed");
```
Comment (tbwebb22) [inline contracts/TokenClaim.sol:75]: Since we're moving to custom error messages instead of require statements, want to go ahead and convert these?

**Review by tbwebb22 — COMMENTED** [review]
Looks good! Just have the comment on using custom error messages instead of the require statements here

File: contracts/TokenClaim.sol (hunk 1 above)
Comment (tbwebb22) [reply contracts/TokenClaim.sol:75]: Sure, just opened it here: https://github.com/decent-dao/fractal-contracts/pull/40

**Review by tbwebb22 — APPROVED** [review]
LGTM!

**Review by sethhrbek — APPROVED** [review]
looks great!

File: test/Token-Claim.test.ts [hunk 2]
```diff
@@ -149,4 +152,24 @@ describe("VotesToken Claiming", function () {
...
+      "DeadlinePending()"
+    );
+  });
+
+  it("Should allow an expired reclaim", async () => {
+    await time.advanceBlocks(5);
+    await tokenClaim.connect(deployer).reclaim();
+    expect(await cToken.balanceOf(deployer.address)).to.eq(
+      ethers.utils.parseUnits("100", 18)
+    );
```
[the remark this answers is not in the document]
Comment (herbig) [reply test/Token-Claim.test.ts:173]: oh cool, @tbwebb22 another option we can start using

**Pull request merged**